src/Assemble.cpp

```diff
@@ -0,0 +1,296 @@
+                fe_values.JxW(q);
+          }
+        }
+      }
+
+      cell->get_dof_indices(dof_indices);
+
+      system_matrix.add(dof_indices, cell_matrix);
+    }
+
+    system_matrix.compress(VectorOperation::add);
+  }
+
+  pcout << "  Assembling the right-hand side" << std::endl;
+
+  // Assemble the right-hand side.
+  {
+    Vector<double> cell_rhs(dofs_per_cell);
+
+    std::vector<types::global_dof_index> dof_indices(dofs_per_cell);
+
+    system_rhs = 0.0;
+
+    // Declare a vector and a tensor containing the local value of Neumann
+    // functions.
+    Vector<double> neumann_loc(dim + 1);
+    Tensor<1, dim> neumann_loc_tensor;
+
+    for (const auto &cell : dof_handler.active_cell_iterators()) {
+      if (!cell->is_locally_owned()) continue;
+
+      cell_rhs = 0.0;
+
+      // Boundary integral for Neumann BCs.
+      if (cell->at_boundary()) {
+        for (unsigned int f = 0; f < cell->n_faces(); ++f) {
+          // Check if the face is at a boundary and if it is a Neunmann
+          // boundary.
+          if (cell->face(f)->at_boundary() &&
+              neumann_boundary_functions.count(cell->face(f)->boundary_id())) {
+            fe_face_values.reinit(cell, f);
+
+            // Set the correct time for the Neumann function.
+            Function<dim> *neumann_function =
+                neumann_boundary_functions[cell->face(f)->boundary_id()];
+            neumann_function->set_time(time_step * deltat);
+
+            for (unsigned int q = 0; q < n_q_face; ++q) {
+              // Get the local value of the Neumann function.
+              neumann_function->vector_value(fe_face_values.quadrature_point(q),
+                                             neumann_loc);
+              for (unsigned int d = 0; d < dim; ++d)
+                neumann_loc_tensor[d] = neumann_loc[d];
+
+              for (unsigned int i = 0; i < dofs_per_cell; ++i) {
+                // Add the Neumann term.
+                cell_rhs(i) +=
+                    scalar_product(neumann_loc_tensor,
+                                   fe_face_values[velocity].value(i, q)) *
+                    fe_face_values.JxW(q);
+              }
+            }
+          }
+        }
+      }
+
+      cell->get_dof_indices(dof_indices);
+      system_rhs.add(dof_indices, cell_rhs);
+    }
+
+    system_rhs.compress(VectorOperation::add);
+
+    // Add the term that comes from the old solution.
+    velocity_mass.block(0, 0).vmult_add(system_rhs.block(0),
+                                        solution_owned.block(0));
+  }
+
+  pcout << "  Applying Dirichlet (BC)" << std::endl;
+
+  // Dirichlet boundary conditions.
+  {
+    std::map<types::global_dof_index, double> boundary_values;
+
+    // Since Dirichlet (BC) are not applied to the pressure, create a mask.
+    ComponentMask mask;
+    if constexpr (dim == 2) {
+      mask = ComponentMask({true, true, false});
+    } else {
+      mask = ComponentMask({true, true, true, false});
+    }
```

Comment:
This can be done more elegantly as follows:
```suggestion
    ComponentMask mask(dim + 1, true);
    mask.set(dim, false);
```